pypesto/objective/function.py

```diff
@@ -114,32 +114,6 @@ def has_res(self) -> bool:
     def has_sres(self) -> bool:
         return callable(self.sres) or self.sres is True
 
-    def check_sensi_orders(
-        self,
-        sensi_orders: Tuple[int, ...],
-        mode: str,
-    ) -> bool:
-        if (mode is MODE_FUN and
-            (0 in sensi_orders and not self.has_fun
-             or 1 in sensi_orders and not self.has_grad
-             or 2 in sensi_orders and not self.has_hess)
-            ) or (mode is MODE_RES and
-                  (0 in sensi_orders and not self.has_res
-                   or 1 in sensi_orders and not self.has_sres)
-                  ):
-            return False
-
-        return True
-
-    def check_mode(self, mode: str) -> bool:
-        if mode == MODE_FUN and not self.has_fun:
-            return False
-
-        if mode == MODE_RES and not self.has_res:
-            return False
-
-        return True
-
```

Comment:
Why did you remove it? It is still used in e.g. the aesaraObjective.

Reply:
moved to the base class ObjectiveBase. The functionality of the has_... functions and check_sensi_orders is mutually referencing as a good default, therefore in derived classes it now suffices to replace either.